token/cli/src/main.rs

```diff
@@ -2574,7 +2580,15 @@ fn main() -> Result<(), Error> {
                 exit(1);
             })
         } else {
-            solana_cli_config::Config::default()
+            if let Some(ref config_file) = *solana_cli_config::CONFIG_FILE {
+                if std::fs::metadata(config_file).is_ok() {
+                    solana_cli_config::Config::load(config_file).unwrap_or_default()
+                } else {
+                    solana_cli_config::Config::default()
+                }
+            } else {
+                solana_cli_config::Config::default()
+            }
```

Comment:
I'll be honest, I find that nigh unreadable, but it's your code base so I'll do whatever you like.

Reply:
That's fair, this one was maybe a bit too much. And also, be sure to rebase because some of this logic was modified. If you need help on the rebase let me know.